internal/support/scheduler/v2/application/scheduler/manager.go

```diff
@@ -0,0 +1,131 @@
+//
+// Copyright (C) 2021 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package scheduler
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/edgexfoundry/edgex-go/internal/pkg/v2/utils"
+	"github.com/edgexfoundry/edgex-go/internal/support/scheduler/config"
+	"github.com/edgexfoundry/edgex-go/internal/support/scheduler/v2/infrastructure/interfaces"
+
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/clients/logger"
+
+	queueV1 "gopkg.in/eapache/queue.v1"
+)
+
+type manager struct {
+	ticker                *time.Ticker
+	lc                    logger.LoggingClient
+	config                *config.ConfigurationStruct
+	mutex                 sync.Mutex
+	executorQueue         *queueV1.Queue
+	intervalToExecutorMap map[string]*Executor
+	actionToIntervalMap   map[string]string
+}
+
+func NewManager(lc logger.LoggingClient, config *config.ConfigurationStruct) interfaces.SchedulerManager {
+	return &manager{
+		ticker:                time.NewTicker(time.Duration(config.Writable.ScheduleIntervalTime) * time.Millisecond),
+		lc:                    lc,
+		config:                config,
+		executorQueue:         queueV1.New(),
+		intervalToExecutorMap: make(map[string]*Executor),
+		actionToIntervalMap:   make(map[string]string),
+	}
+}
+
+func (m *manager) StartTicker() {
+	go func() {
+		for range m.ticker.C {
+			m.triggerInterval()
+		}
+	}()
+}
+
+func (m *manager) StopTicker() {
+	m.ticker.Stop()
+}
+
+func (m *manager) triggerInterval() {
+	nowEpoch := time.Now().Unix()
+
+	defer func() {
+		if err := recover(); err != nil {
+			m.lc.Error("trigger interval error : " + err.(string))
+		}
+	}()
+
+	if m.executorQueue.Length() == 0 {
+		return
+	}
+
+	var wg sync.WaitGroup
+
+	for i := 0; i < m.executorQueue.Length(); i++ {
+		if m.executorQueue.Peek().(*Executor) != nil {
```

Comment:
why do you cast here?
`if m.executorQueue.Peek() != nil {`
is enough

Reply:
Removed. The cast is redundant.